pkg/config/config.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+
+package config
+
+import (
+	"strings"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/posflag"
+	"github.com/knadh/koanf/v2"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+)
+
+type GarmConfig struct {
+	Server   string `koanf:"server" validate:"required,url"`
+	Username string `koanf:"username" validate:"required"`
+	Password string `koanf:"password" validate:"required"`
+}
+
+type OperatorConfig struct {
+	MetricsBindAddress     string        `koanf:"metrics_bind_address" validate:"required,hostname_port"`
+	HealthProbeBindAddress string        `koanf:"health_probe_bind_address" validate:"required,hostname_port"`
+	LeaderElection         bool          `koanf:"leader_election"`
+	SyncPeriod             time.Duration `koanf:"sync_period" validate:"required"`
+	WatchNamespace         string        `koanf:"watch_namespace"`
+}
+
+type AppConfig struct {
+	Garm     GarmConfig     `koanf:"garm"`
+	Operator OperatorConfig `koanf:"operator"`
+}
+
+var Config AppConfig
+
+func ReadConfig(f *pflag.FlagSet, configFile string) error {
```

Comment:
`ReadConfig` reads more like _reading the configuration from file_, but we are doing more a _generating of the configuration_ in this case, 

Therefore i would suggest renaming to:
```suggestion
func GenerateConfig(f *pflag.FlagSet, configFile string) error {
```